--- a/vendor/dolphin/Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp
+++ b/vendor/dolphin/Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp
@@ -288,7 +288,8 @@
 
   // bind the same sub-buffer to all stages
   for (u32 index = 1; index <= 4; index++)
-    glBindBufferRange(GL_UNIFORM_BUFFER, index, s_buffer->m_buffer, buffer.second, data_size);
+    glBindBufferRange(GL_UNIFORM_BUFFER, index, s_buffer->m_buffer, buffer.second,
+                      Common::AlignUp(data_size, 16u));
 
   ADDSTAT(g_stats.this_frame.bytes_uniform_streamed, data_size);
 }
